idea-plugin/build.gradle

```diff
@@ -23,7 +23,7 @@ def name = "palantir-java-format"
 intellij {
     pluginName = name
     updateSinceUntilBuild = true
-    version = "IU-202.6397.94"
+    version = "2024.1"
```

Comment:
Will need to check if this is actually true, but I think we need to set this back down to a lower version before we merge - google-java-format [had it at 2021.3](https://github.com/google/google-java-format/commit/84b2c9a2ba382ee7a968518b46afb0d6000ada8d#diff-6c907a8913b20e03404780aeb811db4c391febccde44b69ac482d896864a802aR38).

Reply:
The [docs](https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html#intellij-extension-version) say:

> The version of the IntelliJ Platform IDE that will be used to build the plugin. 

I think this means the SDK, especially given the errors about `LightProjectDescriptor` not existing in the tests. We should check whether this means you need at least this version of IntelliJ to run it (or perhaps it's fine provided you don't use "new" methods?)

Reply:
> especially given the errors about LightProjectDescriptor not existing in the tests.

yes, I think this is the version used for building & testing the plugin

> least this version of IntelliJ to run it 

I think 'since-build' does this. So we are saying that our plugin is compatible with at least 2021.3 (213)

Reply:
Yeah, I think it's probably fine - the only risk is someone later tries to use some very new method. But I don't expect any work on this intellij plugin needed beyond bugfixes...

Reply:
Yeah, the plugin built using 2024.1 works fine with 2023.1.6